### ISSUE TITLE
VoltageShift in MacOs Monterey 

### ISSUE THREAD

**Issue by kelvincht** [sicreative/VoltageShift#84]
Has anybody made a working example of using VoltageShift in MacOs 12?

**Comment by zspherez** [comment]
Yes, you can find it in my repository here https://github.com/zspherez/VoltageShift/tree/master

**Comment by brechtm** [comment]
@zspherez Does this still require firmware modding (#44) or should this just work?

**Comment by zspherez** [comment]
@brechtm I did not do any firmware modding, and it seems to be working properly for me as per attached picture. Just follow the instructions in my repo closely. 
<img width="533" alt="image" src="https://user-images.githubusercontent.com/66233874/151605605-0c675f99-6a6f-4058-87ed-29da94e2cb6e.png">

**Comment by brechtm** [comment]
> @brechtm I did not do any firmware modding, and it seems to be working properly for me as per attached picture. Just follow the instructions in my repo closely. <img alt="image" width="533" src="https://user-images.githubusercontent.com/66233874/151605605-0c675f99-6a6f-4058-87ed-29da94e2cb6e.png">

The README mentions "_After you build the kext, be sure to load it into your EFI._", but I'm not sure what that means exactly. I suspect this might be a Hackintosh thing, so it might not be possible on a real Mac?

I do happen to be using OpenCore Legacy Patcher (OCLP) to be able to run Monterey on my 2014 MacBook, however. The OCLP partition has an `EFI/OC/Kexts` directory, so I'll try dropping the VoltageShift kext in there...

**Comment by brechtm** [comment]
I managed to install the VoltageShift kext by copying it to the `EFI/OC/Kexts` directory and adding an entry for it (using ProperTree) to `EFI/OC/config.plist`.

Now I can use voltageshift to dis/enable turbo boost and set power limits, but setting a voltage offset doesn't seem to work:

```
 ~/Tools/VoltageShift master 🐚 ./build/Release/voltageshift offset -60
--------------------------------------------------------------------------
VoltageShift offset Tool
--------------------------------------------------------------------------
Before CPU voltageoffset: 0mv
--------------------------------------------------------------------------
After CPU voltageoffset: 0mv
--------------------------------------------------------------------------
```

EDIT: Looks like this still requires a firmware mod (#82). OpenCore might be able to patch the firmware in memory though, since it mentions "_Enables non-invasive, non-permanent firmware updates by replacing protocols in memory"_.

**Comment by zspherez** [comment]
That's interesting. I use VoltageShift on a hackintosh, and often forget that it is used on traditional macs as well. I'm surprised that OCLP does not enable the same firmware updates that required no additional configuration on my end from what I remember. My laptop doesn't allow undervolting through the BIOS, but the offset applied via VoltageShift seems to be working properly, seen by greatly increased battery life and decreased temperatures.

**Comment by imloualvaro** [comment]
> EDIT: Looks like this still requires a firmware mod (#82Detail). OpenCore might be able to patch the firmware in memory though, since it mentions "_Enables non-invasive, non-permanent firmware updates by replacing protocols in memory"_.

u got success to mod your eeprom after OCLP/Monterey firmware update?

and... have you seen some battery improvement using Monterey?

**Comment by mattsoutherden** [comment]
@zspherez Sorry, I can't raise an issue on your repo so I have to comment here. You mention in your readme update that you've removed the binary because if people compile themselves they can use their own certs. However as far as I can tell, unless one has a full Apple developer account and also applies to be a kext developer, then you still need to disable SIP.  I compiled using my own 'Mac Developer' cert and I cant load the kext with SIP enabled. Am I missing something?

**Comment by zspherez** [comment]
@mattsoutherden thanks for the heads up, I enabled issues on that repo. I personally was able to load it without a full kext developer cert, I have no developer cert other than the free one. I have SIP enabled and it still works like a charm. Are you doing this on a hackintosh or a traditional mac? I don't think it would make much of a difference, if you are able to load other kexts you should be able to load this one. Also, try ensuring that you used the version of the kext you built, rather than any other one you may have had laying around.

**Comment by zspherez** [comment]
Feel free to open an issue on my repo if you'd like to discuss further.

**Comment by xCuri0** [comment]
@zspherez Why does readme state I need to sign the kext while also saying to put it in OpenCore EFI kexts ? OpenCore loads unsigned kexts fine and I'm running an unsigned build of voltageshift.kext (built on another PC) without any issues on Monterey SIP on.
